src/components/Select.js

```diff
@@ -59,6 +60,11 @@ function Select(props) {
     testId,
     __internal__focus
   } = mergedProps;
+
+  if (!Array.isArray(options)) {
+    throw new Error(ERROR_STRINGS.FIELD.NO_OPTIONS("Select"));
```

Comment:
There is a disconnect here between what we are testing and what the error message is saying. The error message is talking about the format of the options array, but we don't validate the format here.
I think it's actually a good idea to validate the format because we rely on it in the code below, e.g.:
```
import { areOptionsValid } from "../utils/component"; // or if you can think of a better place

const mergedProps = mergeProps(props, DEFAULT_PROPS, inheritedProps, {
  ...
  options: areOptionsValid
});
const {..., options} = mergedProps;

if (!options) {
  throw new Error(`Select options should have the following format: [{ label: "option-label", value: "option-value" }, ...]`);
}
```